core/src/main/java/org/apache/gravitino/storage/relational/service/RoleMetaService.java

```diff
@@ -291,11 +259,49 @@ public int deleteRoleMetasByLegacyTimeline(long legacyTimeline, int limit) {
         + securableObjectsCount[0];
   }
 
-  private MetadataObject.Type getType(String type) {
+  private static List<SecurableObject> listSecurableObjects(RolePO po) {
+    List<SecurableObjectPO> securableObjectPOs = listSecurableObjectsByRoleId(po.getRoleId());
+    List<SecurableObject> securableObjects = Lists.newArrayList();
+
+    for (SecurableObjectPO securableObjectPO : securableObjectPOs) {
+      String fullName =
+          MetadataObjectService.getMetadataObjectFullName(
+              securableObjectPO.getType(), securableObjectPO.getMetadataObjectId());
+      if (fullName != null) {
+        securableObjects.add(
+            POConverters.fromSecurableObjectPO(
+                fullName, securableObjectPO, getType(securableObjectPO.getType())));
+      } else {
+        LOG.info(
```

Comment:
Can you please change to the warning log.

Reply:
Besides, looks like if the object is deleted, then we will always get null result, maybe we should also delete the relation to role when we delete the object.

Reply:
OK.

Reply:
> Besides, looks like if the object is deleted, then we will always get null result, maybe we should also delete the relation to role when we delete the object.

I have created an issue https://github.com/apache/gravitino/issues/5029 to track it.